### DISCUSSION TITLE
Weird translation of custom sql in `block_on`

### DISCUSSION THREAD

**Discussion by dcaseykc** [moj-analytical-services/splink#1820] [Q&A] [answered]
I'm trying to write a custom set of blocking rules using `block_on`'s sql functionality. However, the translation seems a bit off. I was expecting something like:

`(l.gender = r.gender) OR (r.surname = l.occupation)`. Instead the following code returns `(l."gender" = r."gender") OR (l."surname" = l."occupation" = r."surname" = r."occupation")` and I'm not sure where the extra `=` statements are coming from. 

```python

import pandas as pd
from splink.duckdb.linker import DuckDBLinker
from splink.duckdb.blocking_rule_library import block_on
import splink.duckdb.blocking_rule_library as brl
from splink.datasets import splink_datasets

pd.options.display.max_rows = 1000
df = splink_datasets.historical_50k
settings = {"link_type": "dedupe_only",
            "unique_id_column_name": 'unique_id'}

linker = DuckDBLinker(df, settings)

rules = brl.or_(
    block_on(['gender']),
    brl.and_(
        block_on(['r.surname = l.occupation'])
    )
)

print(rules.blocking_rule_sql)
#> (l."gender" = r."gender") OR (l."surname" = l."occupation" = r."surname" = r."occupation")
```

**Comment by dcaseykc** [comment, marked as answer]
Nevermind. There is a decent chance I just stumbled into the problem described here: https://github.com/moj-analytical-services/splink/issues/1568

(That is, don't use block on -- just pass the sql as a string)

**Reply by RobinL** [reply]
Yeah - that's right.  Sorry for the confusion
